app-server/app/src/main/scala/footprint/Filter.scala

```diff
@@ -0,0 +1,85 @@
+package com.azavea.rf.footprint
+
+import java.sql.Timestamp
+
+import com.lonelyplanet.akka.http.extensions.{PageRequest, Order}
+import geotrellis.slick.Projected
+import geotrellis.vector.Point
+
+import com.azavea.rf.datamodel.latest.schema.tables._
+import com.azavea.rf.datamodel.driver.ExtendedPostgresDriver
+import com.azavea.rf.utils.queryparams._
+import com.azavea.rf.utils.UserErrorException
+
+object FootprintFilters {
+  import ExtendedPostgresDriver.api._
+
+  type FootprintsQuery = Query[Footprints, Footprints#TableElementType, Seq]
+
+  val datePart = SimpleFunction.binary[String, Option[Timestamp], Int]("date_part")
+
+  implicit class FootprintDefault[M, U, C[_]](footprints: FootprintsQuery) {
+    def filterByOrganization(orgParams: OrgQueryParameters): FootprintsQuery = {
+      footprints.filter{ footprint =>
+        orgParams.organization.map(footprint.organizationId === _)
+          .reduceLeftOption(_ || _).getOrElse(true:Rep[Boolean])
+      }
+    }
+
+    def filterByTimestamp(timeParams: TimestampQueryParameters): FootprintsQuery = {
+      footprints.filter{ footprint =>
+        List(
+          timeParams.minCreateDatetime.map(footprint.createdAt > _),
+          timeParams.maxCreateDatetime.map(footprint.createdAt < _),
+          timeParams.minModifiedDatetime.map(footprint.modifiedAt > _),
+          timeParams.maxModifiedDatetime.map(footprint.modifiedAt < _)
+        ).collect({case Some(criteria) => criteria}).reduceLeftOption(_ && _)
+          .getOrElse(true:Rep[Boolean])
+      }
+    }
+
+    def filterByFootprintParams(footprintParams: FootprintQueryParameters): FootprintsQuery = {
+      (footprintParams.x, footprintParams.y) match {
```

Comment:
I think it makes sense to add bounding box (X(min/max) and Y(min/max)) as query parameters since we are going to need it almost immediately.